x-pack/solutions/observability/plugins/synthetics/server/routes/settings/private_locations/helpers.ts

```diff
@@ -60,3 +63,40 @@ export const toSavedObjectContract = (location: PrivateLocation): PrivateLocatio
     spaces: location.spaces,
   };
 };
+
+// This should be called when changing the label of a private location because the label is also stored
+// in the locations array of monitors attributes
+export const updatePrivateLocationMonitors = async ({
+  locationId,
+  newLocationLabel,
+  monitorConfigRepository,
+}: {
+  locationId: string;
+  newLocationLabel: string;
+  monitorConfigRepository: MonitorConfigRepository;
+}) => {
+  const { filtersStr } = parseArrayFilters({
+    locations: [locationId],
+  });
+  const monitorsToUpdate = await monitorConfigRepository.findDecryptedMonitors({
+    spaceId: ALL_SPACES_ID,
+    filter: filtersStr,
+  });
+  const updatedMonitors = monitorsToUpdate.map((m) => {
+    const newLocations = m.attributes.locations.map((l) =>
+      l.id !== locationId ? l : { ...l, label: newLocationLabel }
+    );
+    return {
+      id: m.id,
+      // TODO: Type assertion required due to a structural type mismatch between
+      // what's returned by findDecryptedMonitors and what's expected by bulkUpdate.
+      // This pattern is used throughout the codebase when calling bulkUpdate.
+      // A future refactoring should address this type inconsistency at its source.
+      attributes: { ...m.attributes, locations: newLocations } as unknown as MonitorFields,
+    };
+  });
+
+  await monitorConfigRepository.bulkUpdate({
+    monitors: updatedMonitors,
+  });
```

Comment:
this will only update the saved objects but won't update fleet package policies associates, you will need to use SyntheticsMonitorClient